src/src/rx-serial/SerialSBUS.cpp

```diff
@@ -18,22 +21,45 @@ void SerialSBUS::sendLinkStatisticsToFC()
 uint32_t SerialSBUS::sendRCFrameToFC(bool frameAvailable, uint32_t *channelData)
 {
     crsf_channels_s PackedRCdataOut;
-    PackedRCdataOut.ch0 = channelData[0];
-    PackedRCdataOut.ch1 = channelData[1];
-    PackedRCdataOut.ch2 = channelData[2];
-    PackedRCdataOut.ch3 = channelData[3];
-    PackedRCdataOut.ch4 = channelData[4];
-    PackedRCdataOut.ch5 = channelData[5];
-    PackedRCdataOut.ch6 = channelData[6];
-    PackedRCdataOut.ch7 = channelData[7];
-    PackedRCdataOut.ch8 = channelData[8];
-    PackedRCdataOut.ch9 = channelData[9];
-    PackedRCdataOut.ch10 = channelData[10];
-    PackedRCdataOut.ch11 = channelData[11];
-    PackedRCdataOut.ch12 = channelData[12];
-    PackedRCdataOut.ch13 = channelData[13];
-    PackedRCdataOut.ch14 = channelData[14];
-    PackedRCdataOut.ch15 = channelData[15];
+    
+    if (config.GetSerialProtocol() == PROTOCOL_DJI_RS_PRO)
+    {
+        PackedRCdataOut.ch0 = fmap(channelData[0], CRSF_CHANNEL_VALUE_MIN, CRSF_CHANNEL_VALUE_MAX, 352, 1696);
+        PackedRCdataOut.ch1 = fmap(channelData[1], CRSF_CHANNEL_VALUE_MIN, CRSF_CHANNEL_VALUE_MAX, 352, 1696);
+        PackedRCdataOut.ch2 = fmap(channelData[2], CRSF_CHANNEL_VALUE_MIN, CRSF_CHANNEL_VALUE_MAX, 352, 1696);
+        PackedRCdataOut.ch3 = fmap(channelData[3], CRSF_CHANNEL_VALUE_MIN, CRSF_CHANNEL_VALUE_MAX, 352, 1696);
+        PackedRCdataOut.ch4 = fmap(channelData[5], CRSF_CHANNEL_VALUE_MIN, CRSF_CHANNEL_VALUE_MAX, 352, 1696); // Record start/stop and photo
+        PackedRCdataOut.ch5 = fmap(channelData[6], CRSF_CHANNEL_VALUE_MIN, CRSF_CHANNEL_VALUE_MAX, 352, 1696); // Mode
+        PackedRCdataOut.ch6 = fmap(channelData[7], CRSF_CHANNEL_VALUE_MIN, CRSF_CHANNEL_VALUE_MAX, 176,  848); // Recenter and Selfie
```

Comment:
After some testing the ranges for Mode and Recenter switches appear to be swapped compared to the info from Wodie.  This should be confirmed to still work by more users.

Reply:
Seems to work as-is here in the PR, with the swapped endpoints.

Tested with the endpoints from Woodie, and the recenter/selfie channel still works, but when you trigger the selfie mode, the gimbal mode also toggles from whatever its currently set to, and changes to mode 3 momentarily.

So for example, you have the mode channel set in the middle pos, and the gimbal is in mode 2, then you trigger selfie, the mode changes to mode 3 for a sec, then back to mode 2.

With the channel endpoints as they are in the PR, this does not happen, and the mode stays as set.